src/runtime/runtime_nintendoswitch_init.go

```diff
@@ -0,0 +1,138 @@
+// +build nintendoswitch
+// +build gc.conservative gc.leaking
+
+package runtime
+
+import (
+	"unsafe"
+)
+
+const defaultHeapSize = 0x2000000 * 16 // Default by libnx
+const debugInit = false
+
+// Types of config entry
+const (
+	envEntryTypeEndOfList        = 0 // Entry list terminator.
+	envEntryTypeMainThreadHandle = 1 // Provides the handle to the main thread.
+	envEntryTypeOverrideHeap     = 3 // Provides heap override information.
+)
+
+type configEntry struct {
+	Key   uint32
+	Flags uint32
+	Value [2]uint64
+}
+
+var (
+	heapStart = uintptr(0)
+	heapEnd   = uintptr(0)
+	usedRam   = uint64(0)
+	totalRam  = uint64(0)
+	totalHeap = uint64(0)
+)
+
+func GetHeapBase() uintptr {
+	return heapStart
+}
+
+func GetHeapEnd() uintptr {
+	return heapEnd
+}
+
+// GetContextPtr returns the hblauncher context
+func GetContextPtr() uintptr {
+	return context
+}
+
+// GetMainThreadHandle returns the main thread handler if any
+func GetMainThreadHandle() uintptr {
+	return mainThread
+}
```

Comment:
I suppose these are necessary for gonx?
I do not think this should be exposed in the public API of the runtime. If they are really necessary, maybe it would be a better idea to make them unexported and use them from gonx with `//go:linkname`. This makes it clear this is not a regular API but an implementation detail of the TinyGo runtime.

Reply:
Great! I didn't knew I could do that with go:linkname